File: mobile/lib/core/constants/tracked_lifts.dart
enum TrackedLift {
  benchPress('bench_press', 'Bench Press'),
  squat('squat', 'Squat'),
  deadlift('deadlift', 'Deadlift'),
  overheadPress('overhead_press', 'Overhead Press'),
  pullUpWeightedPullUp('pull_up_weighted_pull_up', 'Pull-Up / Weighted Pull-Up');

  const TrackedLift(this.id, this.label);

  final String id;
  final String label;
}

const trackedLiftIds = <String>{
  'bench_press',
  'squat',
  'deadlift',
  'overhead_press',
  'pull_up_weighted_pull_up',
};

bool isTrackedLiftId(String value) => trackedLiftIds.contains(value);

File: backend/src/shared/constants/tracked-lifts.ts
export const trackedLiftIds = [
  "bench_press",
  "squat",
  "deadlift",
  "overhead_press",
  "pull_up_weighted_pull_up"
] as const;

export type TrackedLiftId = (typeof trackedLiftIds)[number];

export type TrackedLift = {
  id: TrackedLiftId;
  name: string;
};

export const trackedLifts: readonly TrackedLift[] = [
  { id: "bench_press", name: "Bench Press" },
  { id: "squat", name: "Squat" },
  { id: "deadlift", name: "Deadlift" },
  { id: "overhead_press", name: "Overhead Press" },
  { id: "pull_up_weighted_pull_up", name: "Pull-Up / Weighted Pull-Up" }
] as const;

export function isTrackedLiftId(value: string): value is TrackedLiftId {
  return trackedLiftIds.includes(value as TrackedLiftId);
}
